dione-hadoop/src/main/scala/com/paypal/dione/hdfs/index/csv/CsvIndexer.scala

```diff
@@ -0,0 +1,76 @@
+package com.paypal.dione.hdfs.index.csv
+
+import com.paypal.dione.hdfs.index.{HdfsIndexer, HdfsIndexerMetadata}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapreduce.lib.input.{FileSplit, LineRecordReader}
+import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl
+import org.apache.hadoop.mapreduce.{JobID, TaskAttemptID, TaskID, TaskType}
+import org.slf4j.LoggerFactory
+
+/**
+ * based on org.apache.spark.sql.execution.datasources.HadoopFileLinesReader
+ */
+
+case class CsvIndexer(file: Path, start: Long, end: Long, conf: Configuration,
+                      delimiter: Char)
+  extends HdfsIndexer[Seq[String]]() {
+
+  var reader: LineRecordReader = _
+  private var curPosition = -1L
+  private var lastPosition = -1L
+
+  override def closeCurrentFile(): Unit = {
+    reader.close()
+  }
+
+  /**
+   * Regular seek. Called once per offset (block).
+   */
+  override def seek(offset: Long): Unit = {
+    // @TODO - need to better optimize. not to re-init on every seek
```

Comment:
So I guess this is only for random fetches, right? The performance will be impractical for batch...

Reply:
but than again, why not just save the reader open? am I missing anything?

Reply:
currently, the requested use-case is only fetches. but sure I would like to solve for both. started just with something that works.
In CSV we have only offset, sub-offset is almost only zero.
let's say we want to read every other row, then we need to understand when it will be better just to skip a row compared to when will we want to "re-init" from a new offset.

Reply:
ok I got it... In the parquet implementation we have basically the same thing, right? I mean just one offset, when the row group is 128MB or more